### ISSUE TITLE
[DynamiciO] Connection closed when navigating next/link

### ISSUE THREAD

**Issue by khuezy** [vercel/next.js#72297]
### Link to the code that reproduces this issue

https://github.com/khuezy/dynamiio_bug

### To Reproduce

0. pnpm install
1. npm run build
2. npm run start
3. go to http://localhost:3000/A and wait until page fully loads (browser spinner stops)
4. Click on "B" and see "connection closed" crash in console

### Current vs. Expected behavior

App shouldn't crash when navigating.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 22.9.0
  npm: 10.8.3
  Yarn: N/A
  pnpm: 9.12.2
Relevant Packages:
  next: 15.0.3-canary.6 // Latest available version is detected (15.0.3-canary.6).
  eslint-config-next: N/A
  react: 19.0.0-rc-4d577fd2-20241104
  react-dom: 19.0.0-rc-4d577fd2-20241104
  typescript: 5.6.3
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

dynamicIO

### Which stage(s) are affected? (Select all that apply)

next start (local), Other (Deployed)

### Additional context

![error](https://github.com/user-attachments/assets/30284a01-acd7-4c6a-a9ae-10bcec606534)

**Comment by patelvivekdev** [comment]
When you add prefetch to Link, the error disappears.

**Comment by khuezy** [comment]
Great observation.

**Comment by khuezy** [comment]
This was fixed in https://github.com/vercel/next.js/pull/72420, available in 15.0.3
